Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx

```diff
@@ -299,7 +299,7 @@ ContourExtractor2DImageFilter<TInputImage>::GenerateDataForLabels()
   {
     allLabels.assign(inputRange.cbegin(), inputRange.cend());
     std::sort(allLabels.begin(), allLabels.end());
-    const LabelsConstIterator last{ std::unique(allLabels.begin(), allLabels.end()) };
+    const LabelsIterator last{ std::unique(allLabels.begin(), allLabels.end()) };
```

Comment:
Did the use of `LabelsConstIterator` really cause a GCC warning? If so, can you please mention which warning it is?

Reply:
https://open.cdash.org/viewBuildError.php?buildid=7093944 includes the following.  If I have misunderstood or if you see a better way, let's do that instead.
```plaintext
In file included from /home/kitware/Dashboards/TestsModules/Filtering/Path/include/itkContourExtractor2DImageFilter.h:318:0,
                 from /home/kitware/Dashboards/TestsModules/Filtering/Path/test/itkContourExtractor2DImageFilterTest.cxx:21:
/home/kitware/Dashboards/Tests/ITK/Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx: In instantiation of 'void itk::ContourExtractor2DImageFilter<TInputImage>::GenerateDataForLabels() [with TInputImage = itk::Image<unsigned char, 2u>]':
/home/kitware/Dashboards/TestsModules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx:52:5:   required from 'void itk::ContourExtractor2DImageFilter<TInputImage>::GenerateData() [with TInputImage = itk::Image<unsigned char, 2u>]'
/home/kitware/Dashboards/TestsModules/Filtering/Path/test/itkContourExtractor2DImageFilterTest.cxx:758:1:   required from here
/home/kitware/Dashboards/TestsModules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx:303:5: error: no matching function for call to 'std::vector<unsigned char, std::allocator<unsigned char> >::erase(const LabelsConstIterator&, std::vector<unsigned char, std::allocator<unsigned char> >::iterator)'
     allLabels.erase(last, allLabels.end());
     ^
/home/kitware/Dashboards/TestsModules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx:303:5: note: candidates are:
In file included from /usr/include4.8/vector:69:0,
                 from /home/kitware/Dashboards/TestsModules/Core/Common/include/itkConstShapedNeighborhoodIterator.h:21,
                 from /home/kitware/Dashboards/TestsModules/Filtering/Path/test/itkContourExtractor2DImageFilterTest.cxx:19:
/usr/include4.8/bits/vector.tcc:134:5: note: std::vector<_Tp, _Alloc>::iterator std::vector<_Tp, _Alloc>::erase(std::vector<_Tp, _Alloc>::iterator) [with _Tp = unsigned char; _Alloc = std::allocator<unsigned char>; std::vector<_Tp, _Alloc>::iterator = __gnu_cxx::__normal_iterator<unsigned char*, std::vector<unsigned char, std::allocator<unsigned char> > >; typename std::_Vector_base<_Tp, _Alloc>::pointer = unsigned char*]
     vector<_Tp, _Alloc>::
     ^
/usr/include4.8/bits/vector.tcc:134:5: note:   candidate expects 1 argument, 2 provided
/usr/include4.8/bits/vector.tcc:146:5: note: std::vector<_Tp, _Alloc>::iterator std::vector<_Tp, _Alloc>::erase(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::iterator) [with _Tp = unsigned char; _Alloc = std::allocator<unsigned char>; std::vector<_Tp, _Alloc>::iterator = __gnu_cxx::__normal_iterator<unsigned char*, std::vector<unsigned char, std::allocator<unsigned char> > >; typename std::_Vector_base<_Tp, _Alloc>::pointer = unsigned char*]
     vector<_Tp, _Alloc>::
     ^
/usr/include4.8/bits/vector.tcc:146:5: note:   no known conversion for argument 1 from 'const LabelsConstIterator {aka const __gnu_cxx::__normal_iterator<const unsigned char*, std::vector<unsigned char, std::allocator<unsigned char> > >}' to 'std::vector<unsigned char, std::allocator<unsigned char> >::iterator {aka __gnu_cxx::__normal_iterator<unsigned char*, std::vector<unsigned char, std::allocator<unsigned char> > >}'
In file included from /home/kitware/Dashboards/TestsModules/Filtering/Path/include/itkContourExtractor2DImageFilter.h:318:0,
                 from /home/kitware/Dashboards/TestsModules/Filtering/Path/test/itkContourExtractor2DImageFilterTest.cxx:21:
/home/kitware/Dashboards/TestsModules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx:365:63: error: invalid initialization of reference of type 'const itk::ContourExtractor2DImageFilter<TInputImage>::GenerateDataForLabels() [with TInputImage = itk::Image<unsigned char, 2u>]::BoundingBoxType&' from expression of type '<brace-enclosed initializer list>'
       const BoundingBoxType & bbox{ labelBoundingBoxes[label] };
```

Reply:
Ah, thanks @Leengit so now I understand, the `std::vector` implementation of GCC 4.8 does not yet have the C++11 `erase(const_iterator)` overload.

Could you possibly adjust the commit message, to clarify that it also addresses errors, not just warnings? For example (remaining at <= maximum 72 characters):

> COMP: Fix gcc4.8 warnings + errors itkContourExtractor2DImageFilter.hxx

Reply:
I changed the commit message.